src/main/java/wiresegal/wob/arcanum/Parsing.kt

```diff
@@ -167,6 +173,78 @@ fun about(message: Message) {
     })
 }
 
+class Progress(title: String, percentage: String, link: String?) {
```

Comment:
You can turn this into a (private) [data class](https://kotlinlang.org/docs/reference/data-classes.html) and you get the fields as well as the destructuring methods for free.